taxi/views.py

```diff
@@ -87,3 +89,42 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    form_class = DriverCreationForm
+    template_name = "taxi/driver_form.html"
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverLicenseUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+
+    def get_success_url(self):
+        return reverse(
+            "taxi:driver-detail", args=(self.get_context_data()["object"].id,)
+        )
+
+
+@login_required
```

Comment:
You can put this logic into `post` method of `CarDetailView`. The reason is that we don't need to create extra views and url endpoints to handle adding/deleting driver.
So the implementation will look something like this:

1. Add form on car detail template instead of href's that will send value 'Add' or 'Delete' to post method to the cardetailview.
2. You override post method in this view and catch those values that you received in if condition and add/remove driver to car through related fields.
3. You will need to figure out how to get car and user in the view's post method to set related fields.